graphql/fragments/challenge-provider/challengeInfo.graphql

```diff
@@ -25,6 +25,10 @@ fragment ChallengeInfo on Challenge {
       name
       uri
     }
+    location {
```

Comment:
query the `id` field of every nested object - the cache is matching objects by it.